### PR TITLE
First pass at replacing the "layout" approach with a much simpler claim names array one

### PR DESCRIPTION
**Pull request by quartzjer**
This is the simplest possible structure I can imagine to accomplish the minimal goals of a JPT.

It lacks the ability to express more complex JSON structures that internally contain payloads, but this tradeoff may be worth that limitation in order to have a very simple solution.

For more complex requirements that this may not satisfy there are multiple other approaches we can document and suggest, such as using multiple JWPs (the signer can link them using a common claim instance value) and using CBOR/CWPs instead.

### CODE REVIEW

**Review by selfissued — APPROVED** [review]
I believe this simplification aligns with @OR13 's feedback.

File: draft-jmiller-json-proof-token.md [hunk 1]
```diff
@@ -87,51 +87,43 @@ In order to generate a variety of ZKPs of knowledge, range, membeship, or other
...
 
-A layout is the symbolic equivalent to the JSON body of a JWT, containing all of the claim names along with any additional structures or fixed metadata.  The individual claim values are not included in the layout and instead are replaced by payload references.
+In order to facilitate this external definition of the claim names, an additional `cid` key is defined with a required digest value calculated as defined here.  This `cid` can be used similar to a `kid` in order to ensure externally resolve and then verify that the correct list of claim names are being used when processing the payloads containing the claim values.
 
-These claim value references can be present in two different types: direct and indirect.  A direct reference is always an array with a single integer value, acting as an index into the array of payloads contained in the JPT.  All payloads referenced directly MUST be the base64url encoding of the UTF-8 representation of a JSON value.
+If there is an associated JWK containing the signing key information, the `claims` key is also registered there as a convenient location for the claim names.
```
Comment (OR13) [inline draft-jmiller-json-proof-token.md:98]: I am against this language... keys should not store claims, it violates both the familiarity goal of the spec.

File: draft-jmiller-json-proof-token.md [hunk 2]
```diff
@@ -87,51 +87,43 @@ In order to generate a variety of ZKPs of knowledge, range, membeship, or other
...
 
-Indirect references are those understood only in the context of supporting applications.  Payloads with indirect references will have content types that vary by the given context and may be textual or binary.  This allows JPTs to include raw values without incurring the penalty of a double-encoding into a JSON-safe string.
+All payloads are claim values and MUST be the base64url encoding of the UTF-8 representation of a JSON value.
```
Comment (OR13) [inline draft-jmiller-json-proof-token.md:100]: I opened https://github.com/json-web-proofs/json-web-proofs/issues/33 to track the object encoding issue seperatly.

File: draft-jmiller-json-proof-token.md (hunk 1 above)
Comment (OR13) [reply draft-jmiller-json-proof-token.md:98]: Also an example and normative language are needed here.

**Pull request merged**